### ISSUE TITLE
[PCL] Fix clang-tidy issues arisen since #4629

### ISSUE THREAD

**Issue by gnawme** [PointCloudLibrary/pcl#5304]
**Describe the bug**

Issues flagged by `clang-tidy` have crept in since issue #4629 due to delays in approving #4636, which performs `clang-tidy` checks in CI.

**Context**

See logs for #4636. This ticket is to fix those issues, using `clang-tidy` on the command line

**Expected behavior**

Flagged errors will be gone, and `clang-tidy` workflow will pass.

**Current Behavior**

`clang-tidy` workflow in CI fails

**To Reproduce**

See #4636 

**Screenshots/Code snippets**

In order to help explain your problem, please consider adding
* screenshots of the GUI issues
* code snippets: [syntax for code](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code) with correct language highlights

**Your Environment (please complete the following information):**

CI

**Possible Solution**

Run `clang-tidy` on command line to auto-fix

**Additional context**

**Comment by mvieth** [comment]
I don't think these issues have crept in since the pr. For example:
```
/__w/pcl/pcl/common/include/pcl/point_representation.h:321:7: error: use '= default' to define a trivial default constructor [modernize-use-equals-default,-warnings-as-errors]
      DefaultPointRepresentation ()
      ^
```
But point_representation.h hasn't changed since then: https://github.com/PointCloudLibrary/pcl/commits/master/common/include/pcl/point_representation.h
Perhaps the newer version of clang-tidy flags more issues?

**Comment by gnawme** [comment]
> Perhaps the newer version of clang-tidy flags more issues?

Oddly, I'm not seeing the same issues flag on my local repo that are shown in CI.

**Comment by mvieth** [comment]
> Oddly, I'm not seeing the same issues flag on my local repo that are shown in CI.

Same version of clang-tidy? Same files tested?

**Comment by mvieth** [comment]
Fixed by #5318 and #5310
